cmake_modules/FindProtobuf.cmake

```diff
@@ -55,15 +56,34 @@ if (Protobuf_FOUND)
 
     get_target_property (target_type protobuf::libprotobuf TYPE)
     if (target_type STREQUAL "STATIC_LIBRARY")
-        set(PROTOBUF_STATIC_LIB protobuf::libprotobuf)
+      set (PROTOBUF_STATIC_LIB protobuf::libprotobuf)
     endif ()
 
     get_target_property (target_type protobuf::libprotoc TYPE)
     if (target_type STREQUAL "STATIC_LIBRARY")
-        set (PROTOC_STATIC_LIB protobuf::libprotoc)
+      set (PROTOC_STATIC_LIB protobuf::libprotoc)
     endif ()
 
-    get_target_property (PROTOBUF_INCLUDE_DIR protobuf::libprotoc INTERFACE_INCLUDE_DIRECTORIES)
+    get_target_property (PROTOBUF_INCLUDE_DIR protobuf::libprotobuf INTERFACE_INCLUDE_DIRECTORIES)
+    if (NOT PROTOBUF_INCLUDE_DIR)
+      set (PROTOBUF_INCLUDE_DIR ${Protobuf_INCLUDE_DIRS})
+      if (NOT PROTOBUF_INCLUDE_DIR)
+        message(FATAL_ERROR "Cannot determine Protobuf include directory.")
+      endif ()
+    endif ()
+  else ()
+    set (PROTOBUF_LIBRARY ${Protobuf_LIBRARIES})
+    set (PROTOBUF_INCLUDE_DIR ${Protobuf_INCLUDE_DIRS})
+    if (NOT PROTOBUF_INCLUDE_DIR)
+      message(FATAL_ERROR "Cannot determine Protobuf include directory.")
+    endif ()
+
+    if (Protobuf_LIBRARIES MATCHES "\\.a$")
```

Comment:
We may be able to use https://cmake.org/cmake/help/latest/variable/CMAKE_STATIC_LIBRARY_SUFFIX.html instead of `.a`.

Reply:
Thanks for looking at that, '.a' has been replaced with CMAKE_STATIC_LIBRARY_SUFFIX